open-api/src/testFixtures/java/org/apache/iceberg/rest/RCKUtils.java

```diff
@@ -0,0 +1,116 @@
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.eclipse.jetty.server.Server;
+
+class RCKUtils {
+  private static final String CATALOG_ENV_PREFIX = "CATALOG_";
+  static final String RCK_REQUIRES_NAMESPACE_CREATE = "rck.requires-namespace-create";
+  static final String RCK_SUPPORTS_SERVERSIDE_RETRY = "rck.supports-serverside-retry";
+
+  private static Server localServer;
+
+  private RCKUtils() {}
+
+  /**
+   * Utility method that allows configuring catalog properties via environment variables.
+   *
+   * <p>Returns a property map for all environment variables that start with <code>CATALOG_</code>
+   * replacing double-underscore (<code>__</code>) with dash (<code>-</code>) and replacing single
+   * underscore (<code>_</code>) with dot (<code>.</code>) to allow for common catalog property
+   * conventions. All characters in the name are converted to lowercase and values are unmodified.
+   *
+   * <p>Examples:
+   *
+   * <pre><code>
+   *     CATALOG_CATALOG__IMPL=org.apache.iceberg.jdbc.JdbcCatalog -> catalog-impl=org.apache.iceberg.jdbc.JdbcCatalog
+   *     CATALOG_URI=jdbc:sqlite:memory: -> uri=jdbc:sqlite:memory:
+   *     CATALOG_WAREHOUSE=test_warehouse     -> warehouse=test_warehouse
+   *     CATALOG_IO_IMPL=org.apache.iceberg.aws.s3.S3FileIO -> io-impl=org.apache.iceberg.aws.s3.S3FileIO
+   * .    CATALOG_JDBC_USER=ice_user     -> jdbc.user=ice_user
+   *
+   * </code></pre>
+   *
+   * @return configuration map
+   */
+  static Map<String, String> environmentCatalogConfig() {
+    return System.getenv().entrySet().stream()
+        .filter(e -> e.getKey().startsWith(CATALOG_ENV_PREFIX))
+        .collect(
+            Collectors.toMap(
+                e ->
+                    e.getKey()
+                        .replaceFirst(CATALOG_ENV_PREFIX, "")
+                        .replaceAll("__", "-")
+                        .replaceAll("_", ".")
+                        .toLowerCase(Locale.ROOT),
+                Map.Entry::getValue,
+                (m1, m2) -> {
+                  throw new IllegalArgumentException("Duplicate key: " + m1);
+                },
+                HashMap::new));
+  }
+
+  static boolean isLocalServer() {
+    return Boolean.parseBoolean(System.getProperty("rck.local", "true"));
+  }
+
+  static void startLocalServer() throws Exception {
+    RESTCatalogServer restServer = new RESTCatalogServer();
+    localServer = restServer.start();
+  }
+
+  static void stopLocalServer() throws Exception {
+    localServer.stop();
+  }
+
+  static RESTCatalog initCatalogClient() {
+    Map<String, String> catalogProperties = Maps.newHashMap();
+    catalogProperties.putAll(RCKUtils.environmentCatalogConfig());
+    catalogProperties.putAll(Maps.fromProperties(System.getProperties()));
```

Comment:
Would it be possible to get non-default properties primarily from actual test instances? That should allow non-Iceberg code to run these tests against multiple backends in the same JVM. Defaulting to system properties is fine, of course, as long as long as tests have a way to provide overrides per test instance. That should also allow parallel execution.

Reply:
For now, I think it makes sense to keep this simple and just allow for environment variable or java property configuration.  That will allow for running this in a CI pipeline and the env support helps with docker images, which I think @jbonofre and @Fokko are helping to get setup.

I don't think we want to allow parallel execution since some of the tests may conflict in their assertions (e.g. table `t1` does not exist, but some other test is using the same named table).

Reply:
We've been using the predecessor of this PR in PyIceberg [here](https://github.com/apache/iceberg-python/blob/f05b1aedee8451d981188adf68be5e8b360a9ca1/dev/docker-compose-integration.yml#L51-L56).

Regarding parallel execution, I do think that it might be nice to add that, but we need to double-check if the tests are unrelated of each other, as @danielcweeks already pointed out. That's something we can solve down the road.

Reply:
Currently the tests can have collisions since they typically create/assert things like namespaces.  I believe there are a number of situations where the statefulness of the service is different than the in-memory isolation we have with other tests.